components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/IdentityStoreImpl.java

```diff
@@ -102,6 +104,97 @@ public IdentityStoreImpl(List<Domain> domains) throws IdentityStoreException {
      */
 
     @Override
+    public boolean isUserExist(List<Claim> userClaims, String domainName) throws IdentityStoreException {
+        boolean userExists = false;
+        Domain domain;
+        try {
+            if (isNullOrEmpty(domainName)) {
+                domainName = getPrimaryDomainName();
+            }
+            domain = getDomainFromDomainName(domainName);
+        } catch (DomainException e) {
+            throw new IdentityStoreException(String.format("Domain %s was not found", domainName));
+        }
+        String domainUserId;
+
+        for (Claim claim : userClaims) {
+            MetaClaimMapping metaClaimMapping;
+            try {
+                metaClaimMapping = domain.getMetaClaimMapping(claim.getClaimUri());
+            } catch (DomainException e) {
+                throw new IdentityStoreException(String.format
+                        ("Invalid domain configuration found for %s domain. No meta claim mappings.", domainName));
+            }
+            if (domain.isClaimSupported(claim.getClaimUri()) &&
+                    metaClaimMapping.isUnique()) {
+                try {
+                    domainUserId = domain.getDomainUserId(claim);
+                    if (domainUserId != null) {
+                        userExists = true;
+                        break;
+                    }
+                } catch (UserNotFoundException | DomainException e) {
```

Comment:
Need to further review exception handling. In case of user not found only UserNotFoundException should be thrown